presentation/src/main/java/org/sagebionetworks/research/presentation/recorder/reactive/ReactiveFileResultRecorder.java

```diff
@@ -33,47 +38,43 @@
 
     private static final String JSON_OBJECT_DELIMINATOR = ",";
 
-    private final MaybeSubject<FileResult> fileResultMaybeSubject;
+    protected final String deliminator;
 
-    private final Gson gson;
+    protected final String end;
 
     protected final File outputFile;
 
-
-    private final String fileMimeType;
+    protected PrintStream outputStream;
 
     protected final String start;
 
-    protected final String end;
-
-    protected final String deliminator;
-
-    private final AtomicBoolean isFirstJsonObject = new AtomicBoolean(true);
-
-    public static ReactiveFileResultRecorder createJsonArrayLogger(@NonNull String identifier,
-            @NonNull Flowable flowableData, @NonNull Gson gson, @NonNull File outputFile) {
-        return new ReactiveFileResultRecorder(identifier, flowableData, gson,
-                outputFile, JSON_MIME_CONTENT_TYPE, JSON_FILE_START, JSON_FILE_END, JSON_OBJECT_DELIMINATOR);
-    }
+    private final CompositeDisposable compositeDisposable;
 
-    private final ConnectableFlowable<?> connectableFlowableData;
+    private final String fileMimeType;
 
-    private final CompositeDisposable compositeDisposable;
+    private final MaybeSubject<FileResult> fileResultMaybeSubject;
 
-    private Instant startTime;
+    private final Gson gson;
```

Comment:
I don't think the file result recorder should have a Gson instance in it, since this enforces that JSON is the format used. I think this should be  one level down in the class hierarchy.

Reply:
Not sure what this means. This is a FileRecorder for <E>. Are you saying we should have a FileRecorder<String> that doesn't have Gson?

Reply:
I though we considered it a use case where the user wanted to use a non-JSON format, so I originally provided ReactiveFileResultRecorder or something like that, and then a DeviceMotionJsonRecorder to do the specific JSON stuff.